es5-persistence/src/main/java/com/netflix/conductor/dao/es5/index/ElasticSearchDAOV5.java

```diff
@@ -158,19 +148,42 @@ public ElasticSearchDAOV5(Client elasticSearchClient, Configuration config, Obje
                 new LinkedBlockingQueue<>());
     }
 
-    private void updateIndexName(Configuration config) {
-        this.logIndexPrefix = config.getProperty("workflow.elasticsearch.tasklog.index.name", "task_log");
-        this.logIndexName = this.logIndexPrefix + "_" + SIMPLE_DATE_FORMAT.format(new Date());
+    @Override
+    public void setup() throws Exception {
+
+        elasticSearchClient.admin().cluster().prepareHealth().setWaitForGreenStatus().execute().get();
 
         try {
-            elasticSearchClient.admin().indices().prepareGetIndex().addIndices(logIndexName).execute().actionGet();
+
+            initIndex();
+            updateIndexName();
+            Executors.newScheduledThreadPool(1).scheduleAtFixedRate(() -> updateIndexName(), 0, 1, TimeUnit.HOURS);
+
+        } catch (Exception e) {
+            logger.error(e.getMessage(), e);
+        }
+
+        //1. Create the required index
+        try {
```

Comment:
Consider moving this to a separate function. In my opinion, it should help in making the code more testable, increase readability and would make it easy to change later.

Reply:
I will see what I can do. I wanted to keep the PR simple just updating the minimum parts so the contributors were free to refactor and not having to rebase several times if they were working on some feature related to elastic search.